apps/designer/app/shared/theme/theme.ts

```diff
@@ -0,0 +1,84 @@
+import { useLoaderData } from "@remix-run/react";
+import { darkTheme } from "~/shared/design-system";
+import type { ColorScheme, ThemeSetting } from "./types";
+
+// User selected theme setting.
+let setting: ThemeSetting = "dark";
+// Current systeme theme.
+let system: ColorScheme;
+
+/**
+ * Subscribes color scheme change event and rerenders
+ */
+const subscribeSystemTheme = () => {
+  if (typeof matchMedia === "undefined") return;
+  const query = matchMedia("(prefers-color-scheme: light)");
+  const queryColorScheme = (query: MediaQueryList | MediaQueryListEvent) =>
+    query.matches ? "light" : "dark";
+
+  system = queryColorScheme(query);
+
+  query.addEventListener("change", (queryEvent) => {
+    system = queryColorScheme(queryEvent);
+    renderThemeProps();
+  });
```

Comment:
I could not find clean up code. Probably means there is a memory leak. Also, I think this should go into a hook.

Reply:
this is a single event listened binding, its done once and never again

Reply:
I see a function that can be called multiple times potentially :)